api/cluster.go

```diff
@@ -382,12 +388,12 @@ func (s *Server) peerQuerySpeculative(ctx context.Context, data cluster.Traceabl
 			delete(pendingResponses, resp.shardGroup)
 			delete(originalPeers, resp.data.peer.GetName())
 
-		case <-specCheckTicker.C:
+		case <-tickChan:
 			// Check if it's time to speculate!
-			percentReceived := 1 - (float64(len(pendingResponses)) / float64(len(peerGroups)))
-			if percentReceived > speculationThreshold {
+			percentReceived := float64(len(receivedResponses)) / float64(len(peerGroups))
```

Comment:
Is `pendingResponses` still worth anything? It seems superfluous. The current uses are:
1. To drive the main loop (`for len(pendingResponses) > 0`) which could be `for len(receivedResponses) < len(peerGroups)`
2. To decide upon who to speculate which could loop on `peerGroups` and continue if in `recievedResponses`. 

I'm not sure it's worth the extra allocation.